src/main/java/org/broadinstitute/hellbender/engine/ReadWalker.java

```diff
@@ -72,9 +73,11 @@ public void traverse() {
         final CountingReadFilter countedFilter = disable_all_read_filters ?
                                                     new CountingReadFilter("Allow all", ReadFilterLibrary.ALLOW_ALL_READS ) :
                                                     makeReadFilter();
+        final ReadTransformer transformer = makeReadTransformer();
 
         StreamSupport.stream(reads.spliterator(), false)
                 .filter(countedFilter)
+                .map(transformer)
```

Comment:
Here in `ReadWalker` you are applying the transformers after the filters, which is the opposite of the behavior in `LocusWalker`. The behavior should definitely be consistent across the two walker types. We think that the transformers should be applied before the filters in both cases, so that you have an opportunity to repair bad input that would otherwise be filtered out.